src/rules.jl

```diff
@@ -58,10 +58,9 @@
 @define_diffrule Base.acsch(x)                = :( -inv(abs($x) * sqrt(1 + $x^2))      )
 @define_diffrule Base.acoth(x)                = :(  inv(1 - $x^2)                      )
 @define_diffrule Base.sinc(x)                 = :(  cosc($x)                           )
-@define_diffrule Base.deg2rad(x)              = :(  π / 180                            )
-@define_diffrule Base.mod2pi(x)               = :(  isinteger($x / 2pi) ? NaN : 1      )
-@define_diffrule Base.rad2deg(x)              = :(  180 / π                            )
-
+@define_diffrule Base.deg2rad(x)              = :(  oftype($x, π / 180)                )
+@define_diffrule Base.mod2pi(x)               = :(  isinteger($x / 2pi) ? oftype($x, NaN) : one($x) )
```

Comment:
Similarly here, this does not work for general `x`. Maybe use something like
```suggestion
@define_diffrule Base.mod2pi(x)               = :(  isinteger($x / $twoπ) ? oftype(float($x), NaN) : one(float($x)) )
```